cmd/kubeadm/app/apis/kubeadm/v1alpha1/defaults.go

```diff
@@ -142,3 +149,31 @@ func SetDefaultsEtcdSelfHosted(obj *MasterConfiguration) {
 		obj.Etcd.SelfHosted.CertificatesDir = DefaultEtcdCertDir
 	}
 }
+
+// SetDefaults_KubeletConfiguration assigns default values to kubelet
+func SetDefaults_KubeletConfiguration(obj *MasterConfiguration) {
+	if obj.KubeletConfiguration.BaseConfig == nil {
+		obj.KubeletConfiguration.BaseConfig = &kubeletconfigv1alpha1.KubeletConfiguration{}
```

Comment:
If you want all of the KubeletConfiguration fields defaulted, why not just run the defaulter?

Reply:
I didn't check KubeletConfiguration defaulter but I'm afraid that the default values there may not be what we want in kubeadm.

Reply:
Fair enough.

Reply:
We made a compromise here; we first defaulted what kubeadm actually uses, then used the kubelet defaulter for the rest :)